src/EnergyPlus/Pipes.cc

```diff
@@ -317,6 +237,9 @@ namespace Pipes {
 		NumSteamPipes = GetNumObjectsFound( "Pipe:Adiabatic:Steam" );
 		NumLocalPipes = NumWaterPipes + NumSteamPipes;
 		LocalPipe.allocate( NumLocalPipes );
+		for ( int p = 1; p <= NumLocalPipes; ++p ) {
+			LocalPipe( p ) = std::shared_ptr< LocalPipeData >( new LocalPipeData() );
```

Comment:
Welp, if we do the raw pointer storage on the Plant data structure and we just store instances in this array, this should be moot.